File: code_blocks/tools/paywalls_custom_variables_2.swift
import UIKit
import RevenueCat
import RevenueCatUI

class MyViewController: UIViewController {
    func showPaywall() {
        let controller = PaywallViewController()

        // Set custom variables
        controller.setCustomVariable("John", forKey: "player_name")
        controller.setCustomVariable("42", forKey: "level")

        self.present(controller, animated: true)
    }
}

File: code_blocks/tools/paywalls_custom_variables_2.kt
import com.revenuecat.purchases.CustomerInfo
import com.revenuecat.purchases.Offering
import com.revenuecat.purchases.ui.revenuecatui.activity.PaywallActivityLauncher
import com.revenuecat.purchases.ui.revenuecatui.CustomVariableValue

class MyActivity : ComponentActivity() {
    private val paywallActivityLauncher = PaywallActivityLauncher(this)

    fun showPaywall(offering: Offering) {
        paywallActivityLauncher.launch(
            offering = offering,
            customVariables = mapOf(
                "player_name" to CustomVariableValue.String("John"),
                "level" to CustomVariableValue.String("42")
            )
        )
    }
}
